### PR TITLE
feat(presets): ai actions in format bar

### PR DESCRIPTION
**Pull request by pengx17**
refactored some parts so that affine could adapt the ai functions a bit easier.

Actions in format bar is far from complete.
Also some functions like "replace" seems not working properly.

### CODE REVIEW

**Comment by pengx17** [comment]
* **#6679** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/6679?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pengx17 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts [hunk 1]
```diff
@@ -175,10 +189,45 @@ export class AffineAIPanelWidget extends WidgetElement {
...
+      if (prevState === 'hidden') {
+        this._selection = this.host.selection.find('text');
+      } else {
+        // restore selection
+        if (this._selection) {
+          this.host.selection.set([this._selection]);
+        }
+      }
+
+      // tell format bar to show or hide
```
[the remark this answers is not in the document]
Comment (pengx17) [reply packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts:221]: While ai panel is active, we need retain the block/text selection. However doing so the format bar will be displaying. This is a workaround to make sure format bar will not show when ai panel is active.

File: packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts [hunk 2]
```diff
@@ -175,10 +189,45 @@ export class AffineAIPanelWidget extends WidgetElement {
...
+    }
+  };
+
+  protected override willUpdate(changed: PropertyValues): void {
+    const prevState = changed.get('state');
+    if (prevState) {
+      if (prevState === 'hidden') {
+        this._selection = this.host.selection.find('text');
+      } else {
+        // restore selection
```
[the remark this answers is not in the document]
Comment (pengx17) [reply packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts:215]: Will take a look

[the remark this answers is not in the document]
Comment (Flrande) [reply packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts:211]: https://github.com/toeverything/blocksuite/blob/e2f879db8a9636583dd861add3a0765e969404e9/packages/blocks/src/_common/inline/presets/nodes/link-node/link-popup/link-popup.ts#L57

File: packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pengx17) [reply packages/blocks/src/root-block/widgets/ai-panel/ai-panel.ts:220]: Will revisit this issue after this PR

**Review by Flrande — APPROVED** [review]

**Pull request merged**